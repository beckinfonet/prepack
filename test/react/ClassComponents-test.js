/**
 * Copyright (c) 2017-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

/* @flow */

const prepareReactTests = require("./prepareReactTests");
const { runTest } = prepareReactTests();

/* eslint-disable no-undef */
const { it } = global;

it("Simple", async () => {
  await runTest(__dirname + "/ClassComponents/simple.js");
});

it("Simple classes", async () => {
  await runTest(__dirname + "/ClassComponents/simple-classes.js");
});

it("Simple classes #2", async () => {
  await runTest(__dirname + "/ClassComponents/simple-classes-2.js");
});

it("Simple classes #3", async () => {
  await runTest(__dirname + "/ClassComponents/simple-classes-3.js");
});

it("Simple classes with Array.from", async () => {
  await runTest(__dirname + "/ClassComponents/array-from.js");
});

it("Simple classes with Array.from 2", async () => {
  await runTest(__dirname + "/ClassComponents/array-from2.js");
});

it("Inheritance chaining", async () => {
  await runTest(__dirname + "/ClassComponents/inheritance-chain.js");
});

it("Classes with state", async () => {
  await runTest(__dirname + "/ClassComponents/classes-with-state.js");
});

it("Complex class components folding into functional root component", async () => {
  await runTest(__dirname + "/ClassComponents/complex-class-into-functional-root.js");
});

it("Complex class components folding into functional root component #2", async () => {
  await runTest(__dirname + "/ClassComponents/complex-class-into-functional-root2.js");
});

it("Complex class components folding into functional root component #3", async () => {
  await runTest(__dirname + "/ClassComponents/complex-class-into-functional-root3.js");
});

it("Complex class components folding into functional root component #4", async () => {
  await runTest(__dirname + "/ClassComponents/complex-class-into-functional-root4.js");
});

it("Complex class components folding into functional root component #5", async () => {
  await runTest(__dirname + "/ClassComponents/complex-class-into-functional-root5.js");
});
