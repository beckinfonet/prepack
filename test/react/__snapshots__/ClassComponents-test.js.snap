// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classes with state 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Classes with state 2`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Classes with state 3`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Classes with state 4`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Complex class components folding into functional root component #2 1`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #2 2`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #2 3`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #2 4`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #3 1`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #3 2`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #3 3`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #3 4`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #4 1`] = `
ReactStatistics {
  "componentsEvaluated": 12,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #4 2`] = `
ReactStatistics {
  "componentsEvaluated": 12,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #4 3`] = `
ReactStatistics {
  "componentsEvaluated": 12,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #4 4`] = `
ReactStatistics {
  "componentsEvaluated": 12,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #5 1`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #5 2`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #5 3`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component #5 4`] = `
ReactStatistics {
  "componentsEvaluated": 4,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "message": "",
              "name": "Child2",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component 2`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component 3`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Complex class components folding into functional root component 4`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Inheritance chaining 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Inheritance chaining 2`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Inheritance chaining 3`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Inheritance chaining 4`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "MyComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 2`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "MyComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 3`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "MyComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 4`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "MyComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes #2 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes #2 2`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes #2 3`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes #2 4`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes #3 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple classes #3 2`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple classes #3 3`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple classes #3 4`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "NEW_TREE",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple classes 1`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes 2`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes 3`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes 4`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "Child",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 1`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 2`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 2 1`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 2 2`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 2 3`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 2 4`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 3`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;

exports[`Simple classes with Array.from 4`] = `
ReactStatistics {
  "componentsEvaluated": 2,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "",
          "name": "A",
          "status": "INLINED",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 1,
  "optimizedNestedClosures": 1,
  "optimizedTrees": 1,
}
`;
